mev-boost/server/constraints.go

```diff
@@ -95,16 +98,64 @@ func (c *ConstraintCache) AddInclusionConstraints(slot uint64, constraints []*Co
 }
 
 // Get gets the constraints at the given slot.
-func (c *ConstraintCache) Get(slot uint64) (map[common.Hash]*Constraint, bool) {
+func (c *ConstraintCache) Get(slot uint64) (map[gethCommon.Hash]*Constraint, bool) {
 	return c.constraints.Get(slot)
 }
 
 // FindTransactionByHash finds the constraint for the given transaction hash and returns it.
-func (c *ConstraintCache) FindTransactionByHash(txHash common.Hash) (*Constraint, bool) {
+func (c *ConstraintCache) FindTransactionByHash(txHash gethCommon.Hash) (*Constraint, bool) {
 	for _, hashToConstraint := range c.constraints.Values() {
 		if constraint, exists := hashToConstraint[txHash]; exists {
 			return constraint, true
 		}
 	}
 	return nil, false
 }
+
+type (
+	HashToConstraintDecoded = map[gethCommon.Hash]*ConstraintDecoded
+	ConstraintDecoded       struct {
+		Index *uint64
+		Tx    *types.Transaction
+	}
+)
+
+// ParseConstraintsDecoded receives a map of constraints and
+// - creates a slice of constraints sorted by index
+// - creates a slice of constraints without index sorted by nonce and hash
+// Returns the concatenation of the slices
+func ParseConstraintsDecoded(constraints HashToConstraintDecoded) []*ConstraintDecoded {
```

Comment:
can you add a test for this fn? either in the relay or mev-boost is fine

Reply:
Done! Lmk if it's good enough

Reply:
beautiful!